go/arrow/array/dictionary.go

```diff
@@ -1293,6 +1293,29 @@ func (b *BinaryDictionaryBuilder) InsertStringDictValues(arr *String) (err error
 	return
 }
 
+func (b *BinaryDictionaryBuilder) IsNull(i int) bool {
+	switch b := b.idxBuilder.Builder.(type) {
+	case *Uint8Builder:
+		return b.builder.nullBitmap.Len() != 0 && bitutil.BitIsNotSet(b.builder.nullBitmap.Bytes(), i)
+	case *Int8Builder:
+		return b.builder.nullBitmap.Len() != 0 && bitutil.BitIsNotSet(b.builder.nullBitmap.Bytes(), i)
+	case *Uint16Builder:
+		return b.builder.nullBitmap.Len() != 0 && bitutil.BitIsNotSet(b.builder.nullBitmap.Bytes(), i)
+	case *Int16Builder:
+		return b.builder.nullBitmap.Len() != 0 && bitutil.BitIsNotSet(b.builder.nullBitmap.Bytes(), i)
+	case *Uint32Builder:
+		return b.builder.nullBitmap.Len() != 0 && bitutil.BitIsNotSet(b.builder.nullBitmap.Bytes(), i)
+	case *Int32Builder:
+		return b.builder.nullBitmap.Len() != 0 && bitutil.BitIsNotSet(b.builder.nullBitmap.Bytes(), i)
+	case *Uint64Builder:
+		return b.builder.nullBitmap.Len() != 0 && bitutil.BitIsNotSet(b.builder.nullBitmap.Bytes(), i)
+	case *Int64Builder:
+		return b.builder.nullBitmap.Len() != 0 && bitutil.BitIsNotSet(b.builder.nullBitmap.Bytes(), i)
+	default:
+		return false
+	}
+}
```

Comment:
If you just add `IsNull` to the underlying `builder` struct in `builder.go` then *all* the builders will get it for free as they all embed the base builder. It also means you wouldn't need this switch here.

Reply:
That's cool! Let me tackle that!